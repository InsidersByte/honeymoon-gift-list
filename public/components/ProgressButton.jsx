/* @flow */

import React from 'react';
import { RaisedButton } from 'material-ui';

type PropsType = {
    saving: boolean,
};

export default function ProgressButton({ saving, ...props }: PropsType) {
    if (saving) {
        return (
            <RaisedButton
                {...props}
                primary
                label="Saving..."
                type="submit"
                disabled
            />
        );
    }

    return (
        <RaisedButton
            {...props}
            primary
            type="submit"
        />
    );
}
