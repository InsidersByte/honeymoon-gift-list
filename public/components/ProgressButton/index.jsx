/* @flow */

import React from 'react';
import { RaisedButton } from 'material-ui';

type PropsType = {
    saving: boolean,
};

const ProgressButton = ({ saving, ...props }: PropsType) => {
    if (saving) {
        return (
            <RaisedButton
                {...props}
                primary
                label="Saving..."
                type="submit"
                disabled
            />
        );
    }

    return (
        <RaisedButton
            {...props}
            primary
            type="submit"
        />
    );
};

export default ProgressButton;
