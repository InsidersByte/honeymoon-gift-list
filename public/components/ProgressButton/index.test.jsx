import React from 'react';
import { shallow } from 'enzyme';
import ProgressButton from './';

describe('ProgressButton', () => {
  it('should render correctly when not saving', () => {
    const wrapper = shallow(<ProgressButton saving={false} label="Save" />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.prop('label')).toEqual('Save');
    expect(wrapper.prop('disabled')).toEqual(false);
    expect(wrapper.prop('type')).toEqual('submit');
    expect(wrapper.prop('primary')).toEqual(true);
  });

  it('should render correctly when saving', () => {
    const wrapper = shallow(<ProgressButton saving label="Save" />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.prop('label')).toEqual('Saving...');
    expect(wrapper.prop('disabled')).toEqual(true);
    expect(wrapper.prop('type')).toEqual('submit');
    expect(wrapper.prop('primary')).toEqual(true);
  });
});
