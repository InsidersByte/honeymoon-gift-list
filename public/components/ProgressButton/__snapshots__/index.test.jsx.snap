// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressButton should render correctly when not saving 1`] = `
<RaisedButton
  disabled={false}
  fullWidth={false}
  label="Save"
  labelPosition="after"
  primary={true}
  secondary={false}
  type="submit"
/>
`;

exports[`ProgressButton should render correctly when saving 1`] = `
<RaisedButton
  disabled={true}
  fullWidth={false}
  label="Saving..."
  labelPosition="after"
  primary={true}
  secondary={false}
  type="submit"
/>
`;
